cdap-ui/app/cdap/components/CaskWizards/OneStepDeploy/OneStepDeployPlugin.js

```diff
@@ -129,6 +133,7 @@ export default class OneStepDeployPlugin extends Component {
         }
 
         let fetchUrl = `/forwardMarketToCdap?source=${marketPath}&target=${cdapPath}`;
+        fetchUrl += marketHost ? `&marketHost=${encodeURIComponent(marketHost)}` : '';
```

Comment:
Correspondingly we will change the matketPath directly to be full url instead of passing an additional marketHost as query parameter.